cudapoa/samples/sample_cudapoa.cpp

```diff
@@ -203,102 +348,117 @@ int main(int argc, char** argv)
-                if (status == StatusType::success)
-                {
-                    std::cout << "Processed windows " << window_count << " - " << i << std::endl;
+                    // In case that number of batch windows is more than the capacity available on GPU, the for loop breaks into smaller number of windows.
+                    // if adding window i in batch->add_poa_group is not successful, it wont be processed in this iteration, therefore we print i-1
+                    // to account for the fact that window i was excluded at this round.
+                    if (status == StatusType::success)
+                    {
+                        std::cout << "Processed windows " << window_count + window_count_offset << " - " << i + window_count_offset << " (batch " << b << ")" << std::endl;
+                    }
+                    else
+                    {
+                        std::cout << "Processed windows " << window_count + window_count_offset << " - " << i - 1 + window_count_offset << " (batch " << b << ")" << std::endl;
+                    }
                 }
                 else
                 {
-                    std::cout << "Processed windows " << window_count << " - " << i - 1 << std::endl;
+                    // the POA was too large to be added to the GPU, skip and move on
+                    std::cout << "Could not add POA group " << batch_window_ids[i] << "to batch " << b << ". Error code " << status << std::endl;
+                    i++;
                 }
+
+                window_count = i;
             }
-            else
+
+            if (status == StatusType::success)
             {
-                // the POA was too large to be added to the GPU, skip and move on
-                std::cout << "Could not add POA group " << i << " to batch. Error code " << status << std::endl;
+                // Check if all sequences in POA group wre added successfully.
+                for (const auto& s : seq_status)
+                {
+                    if (s == StatusType::exceeded_maximum_sequence_size)
+                    {
+                        std::cerr << "Dropping sequence because sequence exceeded maximum size" << std::endl;
+                    }
+                }
                 i++;
             }
 
-            window_count = i;
-        }
-
-        if (status == StatusType::success)
-        {
-            // Check if all sequences in POA group wre added successfully.
-            for (const auto& s : seq_status)
+            if (status != StatusType::exceeded_maximum_poas && status != StatusType::success)
             {
-                if (s == StatusType::exceeded_maximum_sequence_size)
-                {
-                    std::cerr << "Dropping sequence because sequence exceeded maximum size" << std::endl;
-                }
+                std::cout << "Could not add POA group " << batch_window_ids[i] << "to batch " << b << ". Error code " << status << std::endl;
```

Comment:
Considering that some of the windows with full alignment won't run on most of GPU cards, we can add more details to this error message, like what is the memory requirement and what is the available GPU memory.

Reply:
done.